### PR TITLE
[PM-11689] Incorporate refined submit button capture logic within auto-submit login content script

### PR DESCRIPTION
**Pull request by cagonzalezcs**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11689

## 📔 Objective

The purpose of these changes is to unify the methodology that we use to querying submit button elements between the `OverlayContentService` and the `auto-submit-login.js` content script. This will let us incorporate some of the improvements found when we were re-working the form submission capture behavior for the notification bar.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by jprusik** [comment]
BIT run:
https://github.com/bitwarden/browser-interactions-testing/actions/runs/10744198509

File: apps/browser/src/autofill/utils/index.ts [hunk 1]
```diff
@@ -360,3 +360,42 @@ export function throttle(callback: (_args: any) => any, limit: number) {
...
+  ];
+
+  const keywordsSet = new Set<string>();
+  for (let i = 0; i < keywords.length; i++) {
+    if (typeof keywords[i] === "string") {
+      keywords[i]
+        .toLowerCase()
+        .replace(/[-\s]/g, "")
+        .replace(/[^a-zA-Z0-9]+/g, "|")
+        .split("|")
```
Comment (jprusik) [inline apps/browser/src/autofill/utils/index.ts:391]: Same feedback from 
https://github.com/bitwarden/clients/pull/10617#discussion_r1733302377 and
https://github.com/bitwarden/clients/pull/10617#discussion_r1733306444

```suggestion
        .split(/[^\p{L}\|]+/gu, "|")
```

Comment (cagonzalezcs) [reply apps/browser/src/autofill/utils/index.ts]: Great suggestion, I've applied this regex.

Comment (jprusik) [reply apps/browser/src/autofill/utils/index.ts]: ^ non-blocking, per offline discussion

**Review by jprusik — APPROVED** [review]

**Pull request merged**